docs/package_templates/header_only/all/conanfile.py

```diff
@@ -0,0 +1,117 @@
+    homepage = "https://github.com/project/package"
+    topics = ("topic1", "topic2", "topic3", "header-only") # no "conan"  and project name in topics, keep 'header-only'
+    settings = "os", "arch", "compiler", "build_type" # even for header only
+    no_copy_source = True # do not copy sources to build folder for header only projects
+
+    @property
+    def _minimum_cpp_standard(self):
+        return 14
+
+    # in case the project requires C++14/17/20/... the minimum compiler version should be listed
+    @property
+    def _compilers_minimum_version(self):
+        return {
+            "Visual Studio": "15",
+            "msvc": "14.1",
+            "gcc": "5",
+            "clang": "5",
+            "apple-clang": "5.1",
+        }
+
+    # no exports_sources attribute, but export_sources(self) method instead
+    # this allows finer grain exportation of patches per version
+    def export_sources(self):
+        for p in self.conan_data.get("patches", {}).get(self.version, []):
+            copy(self, p["patch_file"], self.recipe_folder, self.export_sources_folder)
+
+    # same package ID for any package
+    def package_id(self):
+        self.info.clear()
+
+    def requirements(self):
+        # prefer self.requires method instead of requires attribute
+        self.requires("dependency/0.8.1")
+
+    # Only in case the project is header-only
+    def package_id(self):
+        self.info.clear()
+
+    # if another tool than the compiler or CMake is required to build the project (pkgconf, bison, flex etc)
+    def build_requirements(self):
+        self.tool_requires("tool/x.y.z")
+
+    def source(self):
+        # download source package and extract to source folder
+        get(**self.conan_data["sources"][self.version],
+            destination=self.source_folder, strip_root=True)
+
+    def validate(self):
+        # validate the minimum cpp standard supported
+        if self.info.settings.compiler.cppstd:
+            check_min_cppstd(self, self._minimum_cpp_standard)
+        minimum_version = self._compilers_minimum_version.get(str(self.info.settings.compiler), False)
+        if minimum_version and Version(self.info.settings.compiler.version) < minimum_version:
+            raise ConanInvalidConfiguration(f"{self.name} requires C++{self._minimum_cpp_standard}, which your compiler does not support.")
+        # in case it does not work in another configuration, it should validated here too
+        if self.info.settings.os == "Windows":
+            raise ConanInvalidConfiguration(f"{self.ref} can not be used on Windows.")
+
+    def layout(self):
+        # src_folder must use the same source folder name the project
+        basic_layout(self, src_folder="src")
+
+    # not mandatory when there is no patch, but will suppress warning message about missing build() method
+    def build(self):
+        # apply patches in source_folder
+        apply_conandata_patches(self)
```

Comment:
seems wrong where there is `no_copy_source = True`

Reply:
> I would advice to remove no_copy_source = True when patching is required.

Not bad idea at all. 

Indeed, when patching under the build() method, the patch will be applied over source_folder, so it works, however, it looks wrong. I gonna update it and follow your idea.